.gitignore

```diff
@@ -6,6 +6,7 @@ build/*
 *.elf
 *.o
 *.ll
+*.lock
```

Comment:
We should be committing all lock files and updating them as they change.

Reply:
Why _should_ we? It's cargo-generated so it's essentially build output?

Reply:
I have no idea. I have endless battles with endlessly changing lock files, and I've been told to commit them.

Reply:
If anyone can explain, that would be helpful because I don't get it.

Reply:
Ah -- this is news to me. I haven't been committing lock files. I'll start doing so.

Reply:
Ideally @ppannuto or @alevy would chime in, but as I understand it .lock files help with making sure you are using the version of code you expect to be.

I believe that because I use sublime with some rust plugin (compiler errors actually show up right in the .rs files in the editor) that somehow generates .lock files all the time, when otherwise cargo wouldn't update them. So this has been a nuisance for me for years now, and sanity has only returned because I commit them without thinking.

We could switch to deleting them all and gitignoring them. Since they are almost always out of date I really have no idea what value they provide.

Reply:
If anyone needs a refresher like me:
https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html

Basically, _if_ there are any external dependencies, the lock file hones in on a specific SHA for that external crate for you. The Rust rule of thumb seems to be: If your crate is a lib, don't commit locks. Else, your crate must be a bin, so do commit locks. 

While I suppose Tock falls more under the bin category, Tock appears to have an implicit rule of no external dependencies (despite my best efforts to (generate some)[https://github.com/tock/tock/pull/1175] :-P), thus I can't see the value of the lock file for us and at the moment it just generates commit noise.